### ISSUE TITLE
Spotify events not being captured

### ISSUE THREAD

**Issue by Smittttty** [lufinkey/react-native-spotify#115]
I have everything setup, however after adding event listeners, they don't seem to be getting emitted. Am I doing something wrong? Everything else seems to be working correctly with the package. I've tried addListener and also on. I'm currently testing this on iOS.

React native version: 0.59.5
rn-spotify-sdk version: 1.2.6

I've linked both rn-events and also the spotify package

Example code:
```
async componentDidMount() {
    const spotifyOptions = {
      "clientID":"123",
      "sessionUserDefaultsKey":"SpotifySession",
      "redirectURL":"app://auth",
      "scopes":["user-read-private", "playlist-read", "playlist-read-private", "streaming"],
    }

    await Spotify.initialize(spotifyOptions)

    this.setState({
      ready: true
    })

    Spotify.addListener('metadataChange', (data) => {
      console.log("CHANGE")
      console.log(data)
    })

    Spotify.addListener('play', (data) => {
      console.log("PLAY")
      console.log(data)
    })

    Spotify.addListener('pause', (data) => {
      console.log("PAUSE")
      console.log(data)
    })
  }
```

To verify events are working, I wired a TouchableOpacity up to emit the play event, and that was picked yp with the play listener I have above. e.g `onPress={() => Spotify.emit('play', {test: 'test'}}`

**Comment by lufinkey** [comment]
The events only trigger when you're playing something. Have you tried doing that?

**Comment by Smittttty** [comment]
Yup. What I have done is added that componentDidMount into a newly created react-native app's App.js.

I loaded the app received no errors, went to spotify, and started playing a song, and none of my logs are firing with my react-native app closed or open(not sure if this matters).

**Comment by lufinkey** [comment]
Have you tried testing this on Android to see if the issue still occurs?

**Comment by Smittttty** [comment]
We can close this. I was misunderstanding how the library worked!
When I played a song using the Spotify module I started receiving the events. I was under the impression if I played a song using the spotify app, then it would fire the event.

However, I did run into another issue last night. If I'm playing a song using Spotify.playURI, and the song finished. If I try to play that same song again using Spotify.playURI, it doesn't play.

In both cases I'm setting the position to 0, to ensure it should start from the beginning again. It immeditaly fires a metadataChange event, and in that event the status says playing is false.
